tiledb/sm/rest/rest_client.cc

```diff
@@ -102,11 +102,13 @@ Status RestClient::get_array_schema_from_rest(
     const URI& uri, ArraySchema** array_schema) {
   // Init curl and form the URL
   Curl curlc;
-  RETURN_NOT_OK(curlc.init(config_, extra_headers_));
   std::string array_ns, array_uri;
   RETURN_NOT_OK(uri.get_rest_components(&array_ns, &array_uri));
-  std::string url = rest_server_ + "/v1/arrays/" + array_ns + "/" +
-                    curlc.url_escape(array_uri);
+  std::string cache_key = array_ns + ":" + array_uri;
+  RETURN_NOT_OK(curlc.init(
```

Comment:
I was not expecting `cache_key` to be passed into `Curl::init`. This now imposes a restriction that a single `Curl` instance is only good for a single array. Instead, I was expecting this to be an optional parameter to `get_data`, `post_data`, etc. 

Please make this change.

Reply:
You are right. I just thought that since we initialise the Curl instance in every RestClient function , this would not be an issue. I was afraid to change the interface of get_data. But it is totally fine I will do the change.